classes/Product.php

```diff
@@ -6132,6 +6132,54 @@ public function getCustomizationFieldIds()
             WHERE `id_product` = ' . (int) $this->id);
     }
 
+    /**
+     * @return array
+     */
+    public function getNonDeletedCustomizationFieldIds()
```

Comment:
```suggestion
    public function getActiveCustomizationFieldIds()
```

Reply:
Although it sounds better, it might be confused with currently used customization fields. ("Active" could also mean that some product already have customization defined for that field). wdyt?

Reply:
```suggestion
    public function getCustomizationFieldIds($isDeleted = false)
```

?

Reply:
I would agree with it, but there is such method already and I was suggested to avoid modifying existing methods if possible (even when adding optional param) :man_shrugging: Although considering its pretty straightforward method, maybe I can add that optional param or still better a new one? @jolelievre @matks ?

Reply:
If we have to choose between 2 [niladic](https://en.wiktionary.org/wiki/niladic) functions and one monadic (one argument), a lot of [best practices books](https://softwareengineering.stackexchange.com/questions/145055/are-there-guidelines-on-how-many-parameters-a-function-should-accept#:~:text=According%20to%20%22Clean%20Code%3A%20A,function%20is%20zero%20(niladic).) tell us to choose the 2 niladic functions, especially warning us against the use of feature flags like this.

In this theory, one function = one purpose. This boolean would transform existing function `getCustomizationFieldIds()` into a 2-purposes function